### PR TITLE
test: change ceph version of osd on pvc canary test to avoid failures

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

This is to fix the following CI failure.

ci: Canary tests on PVCs are failing when creating OSDs #11600

We set the version of Ceph used for OSD on PVC to v16. It's the moving target and was recently changed the target version. The new version has a regression which fails `ceph-volume raw prepare`. Although this will be fixed in Ceph by the following PR, it's not been backported to Quincy yet.

ceph-volume: fix a bug in lsblk_all()
https://github.com/ceph/ceph/pull/49171

Let's change the version to v17.2.3, the latest version without the above-mentioned regression of Ceph.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [x] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
Perhaps this started failing after Jan 25th when v16.2.11 was released? So if we tested v16.2.10, would it pass? 
Did you also test with v17.2.5 and it has the failure? 

So the cause was a change in the CI host that must have triggered this bug in lsblk_all as called by ceph_volume? 

If the fix is in Ceph main, we should be able to look at our daily CI and see if they would be passing against the latest Ceph main, or not sure if we have a daily test that would hit that issue.

**Review by travisn — APPROVED** [review]
Great to have this fix, I'd suggest to merge so we can get green CI again, and separately we can confirm the fix in main.

**Comment by satoru-takeuchi** [comment]
@travisn 

> Perhaps this started failing after Jan 25th when v16.2.11 was released?

I guess so.

> So if we tested v16.2.10, would it pass?

I verified that OSD on partition-backed PVC creation succeeded in Rook v1.10.10 + Ceph v16.2.10 and failed in Rook v1.10.10 and Ceph v16.2.11.

> Did you also test with v17.2.5 and it has the failure?

Yes. Please see also [this message](https://github.com/rook/rook/pull/11600#issuecomment-1425104380).

> So the cause was a change in the CI host that must have triggered this bug in lsblk_all as called by ceph_volume?
> If the fix is in Ceph main, we should be able to look at our daily CI and see if they would be passing against the latest Ceph main, or not sure if we have a daily test that would hit that issue.

My guess might be wrong. [The ceph issue corresponds to ceph/ceph#49171](https://tracker.ceph.com/issues/58137) describes the reporter said ceph-volume failed due to `KeyError: 'KNAME'`. However, Rook's CI failures hit `KeyError: 'TYPE'`.

In summary, what I can say now is as follows.

- The canary tests for OSD on PVC used `quay.io/ceph/ceph:v16` by Jan 25th.
- Before Jan 25th, v16 pointed to v16.2.10 and after this day, v16 points to v16.2.11.
- The creation of OSD on partition-backed PVC succeeds in Ceph <= v{16.2.10,v17.2.3} and fails in the newer versions.
- This problem *might* be fixed in [this PR](https://github.com/ceph/ceph/pull/49171).

If ceph/ceph#49171 isn't what we wont, we should submit a new Ceph issue.

**Comment by travisn** [comment]
@satoru-takeuchi Your analysis looks correct, I opened #11656 to track it until fixed in Ceph.
